app/decorators/pseud_decorator.rb

```diff
@@ -42,7 +43,7 @@ def works_count
   end
 
   def bookmarks_count
-    data[:public_bookmarks_count]
+    User.current_user.present? ? data[:general_bookmarks_count] : data[:public_bookmarks_count]
```

Comment:
Should we do `count || 0` like works_count in case of nil?

Reply:
I'm not really sure!

@elzj was there a reason the `|| 0` was necessary for works but not bookmarks in the initial code? Or was it just an oversight?

Reply:
That was #3207. I don't know if it's different for bookmarks.

Reply:
It looks like if a user has no bookmarks, both the general and public bookmark count are properly set to 0 instead of nil and no 500 error occurs:

```
"_index": "ao3_development_pseuds",
        "_type": "pseud",
        "_id": "318811",
        "_score": 1,
        "_source": {
          "id": 318811,
          "user_id": 293707,
          "name": "testy4",
          "description": null,
          "created_at": "2014-02-25T05:10:19.000-05:00",
          "user_login": "testy4",
          "byline": "testy4",
          "collection_ids": [],
          "sortable_name": "testy4",
          "fandoms": [
            {
              "id": 6670,
              "name": "Testing",
              "count": 1
            }
          ],
          "general_bookmarks_count": 0,
          "public_bookmarks_count": 0,
          "general_works_count": 1,
          "public_works_count": 1
        }
```

Reply:
I think I see the difference:

- for work counts, we do `pseud.works.where(countable_works_conditions).group(:restricted).count`, get a hash, access the 'false' key for public works, which can be nil
- for bookmark counts, it's always count queries without grouping so we don't have that problem